### PR TITLE
DOC: Updated documentation for scipy.linalg.qr

### LINKED ISSUE #11178: DOC: https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.qr.html
The docs say: "overwrite_a bool, optional. Whether data in a is overwritten (may improve performance)".

But it does not say whether the improved performance is for **overwrite_a=True** or **overwrite_a=False**!

### PR DESCRIPTION
**Pull request by gaurav-1205**
#### Reference issue
 Closes gh-11178

#### What does this implement/fix?
 Clarifies the purpose of param `overwrite_a`

### CODE REVIEW

File: scipy/linalg/decomp_qr.py [hunk 1]
```diff
@@ -37,8 +37,10 @@ def qr(a, overwrite_a=False, lwork=None, mode='full', pivoting=False,
     ----------
     a : (M, N) array_like
         Matrix to be decomposed
-    overwrite_a : bool, optional
-        Whether data in a is overwritten (may improve performance)
+    overwrite_a : bool, optional(default - False)
```
Comment (tylerjereddy) [inline scipy/linalg/decomp_qr.py]: The `False` default is also present/visible in the parameter list at the top, so I'm not sure if we need this addition. If we do add this, will at least need a space before the `(`.

File: scipy/linalg/decomp_qr.py [hunk 2]
```diff
@@ -37,8 +37,10 @@ def qr(a, overwrite_a=False, lwork=None, mode='full', pivoting=False,
...
+        Whether data in a is overwritten (may improve performance if overwrite_a 
+        is set True by reusing the existing input data structure rather than
```
Comment (tylerjereddy) [inline scipy/linalg/decomp_qr.py]:
```suggestion
        is set to True by reusing the existing input data structure rather than
```

**Review by tylerjereddy — COMMENTED** [review]
Looks like the doc improvement could be helpful---I added a few small comments.

File: scipy/linalg/decomp_qr.py (hunk 1 above)
Comment (rgommers) [reply scipy/linalg/decomp_qr.py]: agreed, don't need this

**Comment by rgommers** [comment]
Addressed last comment and merged. Thanks @gaurav-1205

**Comment by mdhaber** [comment]
Thanks @gaurav-1205! If you'd like your name added to [THANKS.txt](https://github.com/scipy/scipy/blob/master/THANKS.txt), please make an entry [here](https://github.com/scipy/scipy/wiki/*THANKS.txt-additions-modifications-for-1.5.0*).

**Pull request merged**